broadinstitute/single_cell_portal_core
# Remove plotly filter/group dependency (SCP-3756)

This removes our dependency on the deprecated plotly transform feature. The replacement is reasonably performant (~2-3ms for <10K points, 20-30ms for 100K points, and 200ms for 1MM points) and enables finer control over plot point ordering.  Subsequent [analysis](https://docs.google.com/document/d/1JKmXp9gEY2Y_cshypBwLrpgBEKYcpBmMna0nf_7eeNw/edit) shows this is substantially faster than using plotly's filtering. 
 
Accordingly, this PR includes the demoed feature of bringing to the front clusters that are being hovered over. This PR also includes code for filtering points by expression value, as it was useful for testing the filter code, however that code is turned off pending more UX refinement.

this also refactors a lot of the visualization code to enable hot reloads.

TO TEST:

load a cluster plot
confirm the clusters/labels appear correctly
confirm hovering over a label brings the points of that label to the front
confirm all other scatter plot visualizations (expression plot, annotated scatter, correlated scatter) still work too
